libs/parallelism/algorithms/include/hpx/parallel/algorithms/transform_inclusive_scan.hpp

```diff
@@ -186,330 +578,249 @@ namespace hpx { namespace parallel { inline namespace v1 {
+                std::forward<Conv>(conv), std::move(init),
                 std::forward<Op>(op));
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
+    }
+
+    template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+        typename Conv, typename Op,
+        HPX_CONCEPT_REQUIRES_(hpx::is_execution_policy<
+            ExPolicy>::value&& hpx::traits::is_iterator<FwdIter1>::value&&
+                hpx::traits::is_iterator<FwdIter2>::value&& hpx::is_invocable_v<
+                    Conv, typename std::iterator_traits<FwdIter1>::value_type>&&
+                    hpx::is_invocable_v<Op,
+                        typename hpx::util::invoke_result<Conv,
+                            typename std::iterator_traits<
+                                FwdIter1>::value_type>::type,
+                        typename hpx::util::invoke_result<Conv,
+                            typename std::iterator_traits<
+                                FwdIter1>::value_type>::type>)>
+    // clang-format on
+    HPX_DEPRECATED_V(1, 8,
+        "hpx::parallel::transform_inclusive_scan is deprecated, use "
+        "hpx::transform_inclusive_scan instead")
+        typename util::detail::algorithm_result<ExPolicy, FwdIter2>::type
+        transform_inclusive_scan(ExPolicy&& policy, FwdIter1 first,
+            FwdIter1 last, FwdIter2 dest, Op&& op, Conv&& conv)
+    {
+        static_assert(hpx::traits::is_forward_iterator<FwdIter1>::value,
+            "Requires at least forward iterator.");
+        static_assert(hpx::traits::is_forward_iterator<FwdIter2>::value,
+            "Requires at least forward iterator.");
+
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+        return hpx::parallel::v1::detail::transform_inclusive_scan<FwdIter2>()
+            .call(std::forward<ExPolicy>(policy), first, last, dest,
+                std::forward<Conv>(conv), std::forward<Op>(op));
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
+    }
+}}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::transform_inclusive_scan
+    HPX_INLINE_CONSTEXPR_VARIABLE struct transform_inclusive_scan_t final
+      : hpx::functional::tag_fallback<transform_inclusive_scan_t>
+    {
+        // clang-format off
+        template <typename InIter, typename OutIter, typename BinOp,
+            typename UnOp,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter>::value &&
+                hpx::traits::is_iterator<OutIter>::value &&
+                hpx::is_invocable_v<UnOp,
+                    typename std::iterator_traits<InIter>::value_type> &&
+                hpx::is_invocable_v<BinOp,
+                    typename hpx::util::invoke_result<UnOp,
```

Comment:
FWIW, we do have `hpx::invoke_result_t<>` (https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/functional/include/hpx/functional/invoke_result.hpp#L41-L42)